Lib/zipimport.py

```diff
@@ -320,6 +321,29 @@ def get_resource_reader(self, fullname):
         from importlib.readers import ZipReader
         return ZipReader(self, fullname)
 
+    def _get_files(self):
+        """Return the files within the archive path."""
+        try:
+            mtime = _bootstrap_external._path_stat(self.archive).st_mtime
+        except OSError:
+            mtime = -1
+        if mtime != self._archive_mtime:
+            _zip_directory_cache[self.archive] = _read_directory(self.archive)
+            self._archive_mtime = mtime
+
+        try:
+            files = _zip_directory_cache[self.archive]
+        except KeyError:
+            files = _read_directory(self.archive)
+            _zip_directory_cache[self.archive] = files
+            self._archive_mtime = mtime
+
+        self._files = files
+        return self._files
+
+    def invalidate_caches(self):
+        """Invalidate the mtime of the ZIP archive."""
+        self._archive_mtime = -1
```

Comment:
Is there a specific reason to add the stat call on every access of the importer instead of keeping the cache as-is and instead rebuilding it when `invalidate_caches()` is called (i.e. make `invalidate_caches()` take over the part of `__init__()` that sets `self._files`)? Since the common case is going to be no file is changed, I think optimizing performance that way to avoid the stat call makes sense.

You also didn't remove `self._files`, which led to two different sets of file data.

Reply:
In this case I was following the pattern in `FileFinder.find_spec()` where the cache is refreshed whenever the directory that a FileFinder is handling has been modified. Maybe it isn't so applicable because a zip file isn't modified as often.

If we're removing this check for zip file modification and consolidating setting `self._files` in `invalidate_caches()`, then it also makes sense to revert `_get_files()` back to `self._files` since no additional checks are being done on accessing file data. Keeping track of the zip file's mtime probably isn't needed anymore either since a user presumably always wants to call `_read_directory()` if `invalidate_caches()` is being called.